src/main/java/Parser.java

```diff
@@ -0,0 +1,206 @@
+/**
+ * Simulates a system that parses user input into
+ * commands for the Duke System.
+ * @author Fabian Chia Hup Peng
+ */
+
+public class Parser {
+
+    /**
+     * Parsers user input into appropriate commands.
+     * @param fullCommand The user input in String format.
+     * @return The command to be carried out by Duke.
+     */
+    public static Command parse(String fullCommand) {
+
+        Command returnCommand = new DummyCommand();
```

Comment:
What about just declaring the `returnCommand` variable here without instantiating the `DummyCommand`? Then you can throw an exception in the `else` case if `returnCommand` is not assigned to anything. In this way, you can get rid of the dummy class.